### ISSUE TITLE
google_sql_user : Valid Service Accounts exceed 63 character limit 

### ISSUE THREAD

**Issue by parag-warudkar-ck** [hashicorp/terraform-provider-google#15520]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

1.3.6

### Affected Resource(s)

* google_sql_database_instance - POSTGRESQL_15
* google_sql_user

### Terraform Configuration Files

```tf
resource "google_sql_user" "sql_iam_user" {
  for_each = { for x in local.sql_iam_users : "${x.project}::${x.instance}::${x.name}" => x if local.sql_iam_users != null } 
  project  = each.value.project
  instance = each.value.instance
  name     = trimsuffix(each.value.name, ".gserviceaccount.com")
  type     = "CLOUD_IAM_SERVICE_ACCOUNT"
}
```

### Debug Output

Terraform v1.3.6
on linux_amd64

│ Error: Error, failed to insert user aaaa-bb-cccccccc-dddddd-eeeeee@ffffff-de-gggggggg-hhhh-iiiiii.iam into instance blah: googleapi: Error 400: Invalid request: User name "aaaa-bb-cccccccc-dddddd-eeeeee@ffffff-de-gggggggg-hhhh-iiiiii.iam" to be created is too long (max 63).., invalid
│ 

### Expected Behavior

User should have been added. 
### Actual Behavior
The service account is valid but the truncation without gserviceaccount.com still leaves it with 65 characters and results in 400 error too long as provided above.

This means for a perfectly valid service account there is no way to use IAM authentication.

### Steps to Reproduce

1. Create a long yet still valid service account
2. Try to add it as an CLOUD_IAM_SERVICE_ACCOUNT with google_sql_user 

### References

* #0000

b/296623359

**Comment by ShubhamAvasthi** [comment]
I am from the Cloud SQL team at Google. We discussed internally and decided to not increase the 63-character limit for Postgres users. More details are available in the internal ticket b/296623359.

**Comment by roaks3** [comment]
Agreed that this makes sense to leave the way we have it, and the existing error message seems sufficient (ie. the service account may be valid, but its name needs to be shortened for use with Postgres).

Closing as Working as Expected.
